H2MLauncher.UI/ViewModels/SettingsViewModel.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.IO;
+using System.Windows.Input;
+
+using Awesome.Net.WritableOptions;
+
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+
+using H2MLauncher.Core.Settings;
+using H2MLauncher.UI.Dialog;
+
+using Microsoft.Win32;
+
+namespace H2MLauncher.UI.ViewModels;
+
+public partial class SettingsViewModel : DialogViewModelBase
+{
+    [ObservableProperty]
+    private string _mwrLocation = "";
+
+    [ObservableProperty]
+    private string _iw4mMasterServerUrl = "";
+
+    public ICommand ApplyCommand { get; set; }
+
+    public ICommand CancelCommand { get; set; }
+
+    public SettingsViewModel(IWritableOptions<H2MLauncherSettings> options)
+    {
+        MwrLocation = options.Value.MWRLocation;
```

Comment:
Check for nulls

Reply:
I cannot be null, defaults to empty string

Reply:
I mean the option object